From: James Bottomley <james.bottomley@steeleye.com>
Subject: sas transport sysfs write attributes bug fix
References: 166035
Patch-Mainline: yes

None of the SAS Transport attributes can be written to.

Signed-off-by: Eric Moore <emoore@lsil.com>
Signed-off-by: Hannes Reinecke <hare@suse.de>

diff -uarN b/drivers/scsi/scsi_transport_sas.c a/drivers/scsi/scsi_transport_sas.c
--- b/drivers/scsi/scsi_transport_sas.c	2006-04-26 12:18:27.000000000 -0600
+++ a/drivers/scsi/scsi_transport_sas.c	2006-04-26 12:56:21.000000000 -0600
@@ -977,7 +977,6 @@
 #define SETUP_TEMPLATE(attrb, field, perm, test)				\
 	i->private_##attrb[count] = class_device_attr_##field;		\
 	i->private_##attrb[count].attr.mode = perm;			\
-	i->private_##attrb[count].store = NULL;				\
 	i->attrb[count] = &i->private_##attrb[count];			\
 	if (test)							\
 		count++
